third_party/txt/src/txt/paragraph.cc

```diff
@@ -919,6 +919,10 @@ void Paragraph::Layout(double width, bool force) {
             [](const CodeUnitRun& a, const CodeUnitRun& b) {
               return a.code_units.start < b.code_units.start;
             });
+
+  if (paragraph_style_.width_type == TextWidthType::tight) {
```

Comment:
(see above comments for context) LibTxt should be able to indiscriminately provide the metrics for both width types. By making the paragraph "choose" which width to provide, we are artificially hiding a potentially useful metric from the users of this quite low level API.

This hiding makes more sense if done completely in the framework level.